csrc/scheduler/heuristic_types.h

```diff
@@ -55,11 +55,13 @@ enum class ScheduleHeuristic {
   InnerPersistent,
   InnerOuterPersistent,
   OuterPersistent,
-  Transpose
+  Transpose,
+  ExprEval
 };
 
 //! Define a schedule table to loop over all the heuristics in priority order.
-constexpr std::array<ScheduleHeuristic, 8> all_heuristics_in_priority_order = {
+constexpr std::array<ScheduleHeuristic, 9> all_heuristics_in_priority_order = {
+    ScheduleHeuristic::ExprEval,
```

Comment:
Should `NoOp` come before `ExprEval`?

Reply:
Some cases get accepted by NoOp scheduler, which is why I prioritized ExprEval scheduler.

We may need to change the heuristics of `NoOp` if we want to switch the order.

Reply:
Oh! Thanks for mentioning that. Does `NoOp` scheduler accept the cases where you have a single scalar output? Because it seems to me that it would do so based on this code: https://github.com/NVIDIA/Fuser/blob/8c18701e252f279292f9ae1c926b7e919b3ebd17/csrc/fusion.cpp#L341-L359
We should add a special case for zero-dimensional outputs there! I'll send a PR in a moment that does that and flips NoOp to be first again.

Reply:
https://github.com/NVIDIA/Fuser/issues/2241

Reply:
```
[  FAILED  ] 6 tests, listed below:
[  FAILED  ] ATenNodesParametrizedTest.MatmulNodeConcrete/2, where GetParam() = ({ 32 }, { 32, 1 })
[  FAILED  ] ATenNodesParametrizedTest.MatmulNodeConcrete/8, where GetParam() = ({ 1, 32 }, { 32 })
[  FAILED  ] ATenNodesParametrizedTest.MatmulNodeConcrete/10, where GetParam() = ({ 1, 32 }, { 32, 1 })
[  FAILED  ] ATenNodesParametrizedTest.MatmulNodeSymbolic/2, where GetParam() = ({ 32 }, { 32, 1 })
[  FAILED  ] ATenNodesParametrizedTest.MatmulNodeSymbolic/8, where GetParam() = ({ 1, 32 }, { 32 })
[  FAILED  ] ATenNodesParametrizedTest.MatmulNodeSymbolic/10, where GetParam() = ({ 1, 32 }, { 32, 1 })
```

It is likely because there no reductions identified since we use ATen, and all the dimensions in the output are broadcast dimensions. So the cases where M/N = 1 get picked by `NoOp`